File: Calculadora.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.IOException;

public class Calculadora {
    public static void main(String[] args) throws Exception {
        // Cargar archivo de entrada
        CharStream input = CharStreams.fromFileName(args[0]);

        // Crear lexer y parser
        Gramatica_CalLexer lexer = new Gramatica_CalLexer(input);
        CommonTokenStream tokens = new CommonTokenStream(lexer);
        Gramatica_CalParser parser = new Gramatica_CalParser(tokens);

        // Obtener árbol
        ParseTree tree = parser.prog();

        // Evaluar con visitor
        CalVisitor eval = new CalVisitor();
        eval.visit(tree);
    }
}

File: Calculadora.py
import sys
from antlr4 import FileStream, CommonTokenStream
from Gramatica_CalLexer import Gramatica_CalLexer
from Gramatica_CalParser import Gramatica_CalParser
from CalVisitor import CalVisitorPython

def main(file):
    input_stream = FileStream(file)
    lexer = Gramatica_CalLexer(input_stream)
    stream = CommonTokenStream(lexer)
    parser = Gramatica_CalParser(stream)
    tree = parser.prog()  # regla inicial

    visitor = CalVisitorPython()

    # recorrer cada stat en prog
    for stat in tree.stat():
        if hasattr(stat, 'expr') and stat.expr() is not None:
            result = visitor.visit(stat.expr())
            print(result)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Uso: python CalculadoraPython.py archivo.expr")
        sys.exit(1)
    main(sys.argv[1])
